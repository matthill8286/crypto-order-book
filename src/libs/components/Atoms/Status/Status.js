import { css, styled } from "../../../styles/styled";
export var Status = styled.div.withConfig({
  displayName: "Status",
  componentId: "y46az8-0"
})(["", ";width:", ";height:", ";background:", ";font-family:", ";font-size:", ";color:", ";border-radius:100%;text-align:center;display:flex;justify-content:center;align-items:center;"], function (_ref) {
  var withinIcon = _ref.withinIcon,
      theme = _ref.theme;
  return withinIcon && css(["position:absolute;margin-left:", ";margin-top:-", ";z-index:90;"], theme.spacing.base.xs, theme.spacing.base.xs);
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.spacing.base.sm;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.spacing.base.sm;
}, function (_ref4) {
  var theme = _ref4.theme,
      _ref4$type = _ref4.type,
      type = _ref4$type === void 0 ? 'primary' : _ref4$type;
  return theme.color[type];
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.font.family.default;
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.font.size.xxs;
}, function (_ref7) {
  var theme = _ref7.theme,
      _ref7$type = _ref7.type,
      type = _ref7$type === void 0 ? 'primary' : _ref7$type;
  return type === 'white' ? theme.color.grey6 : theme.color.white;
});
//# sourceMappingURL=Status.js.map