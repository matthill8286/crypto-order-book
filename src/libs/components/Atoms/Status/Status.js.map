{"version":3,"sources":["../../../../src/components/Atoms/Status/Status.tsx"],"names":["css","styled","Status","div","withinIcon","theme","spacing","base","xs","sm","type","color","font","family","default","size","xxs","grey6","white"],"mappings":"AAAA,SAASA,GAAT,EAAcC,MAAd;AASA,OAAO,IAAMC,MAAM,GAAGD,MAAM,CAACE,GAAV;AAAA;AAAA;AAAA,4LACf;AAAA,MAAGC,UAAH,QAAGA,UAAH;AAAA,MAAeC,KAAf,QAAeA,KAAf;AAAA,SACAD,UAAU,IACVJ,GADU,sEAGOK,KAAK,CAACC,OAAN,CAAcC,IAAd,CAAmBC,EAH1B,EAIOH,KAAK,CAACC,OAAN,CAAcC,IAAd,CAAmBC,EAJ1B,CADV;AAAA,CADe,EASR;AAAA,MAAGH,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,OAAN,CAAcC,IAAd,CAAmBE,EAAlC;AAAA,CATQ,EAUP;AAAA,MAAGJ,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,OAAN,CAAcC,IAAd,CAAmBE,EAAlC;AAAA,CAVO,EAWH;AAAA,MAAGJ,KAAH,SAAGA,KAAH;AAAA,yBAAUK,IAAV;AAAA,MAAUA,IAAV,2BAAiB,SAAjB;AAAA,SAAiCL,KAAK,CAACM,KAAN,CAAYD,IAAZ,CAAjC;AAAA,CAXG,EAYF;AAAA,MAAGL,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACO,IAAN,CAAWC,MAAX,CAAkBC,OAAjC;AAAA,CAZE,EAaJ;AAAA,MAAGT,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACO,IAAN,CAAWG,IAAX,CAAgBC,GAA/B;AAAA,CAbI,EAcR;AAAA,MAAGX,KAAH,SAAGA,KAAH;AAAA,yBAAUK,IAAV;AAAA,MAAUA,IAAV,2BAAiB,SAAjB;AAAA,SACPA,IAAI,KAAK,OAAT,GAAmBL,KAAK,CAACM,KAAN,CAAYM,KAA/B,GAAuCZ,KAAK,CAACM,KAAN,CAAYO,KAD5C;AAAA,CAdQ,CAAZ","sourcesContent":["import { css, styled } from '@/styles/styled'\n\nexport type StatusType = 'white' | 'black' | 'primary'\n\nexport interface StatusProps {\n  type?: StatusType\n  withinIcon?: boolean\n}\n\nexport const Status = styled.div<StatusProps>`\n  ${({ withinIcon, theme }) =>\n    withinIcon &&\n    css`\n      position: absolute;\n      margin-left: ${theme.spacing.base.xs};\n      margin-top: -${theme.spacing.base.xs};\n      z-index: 90;\n    `};\n  width: ${({ theme }) => theme.spacing.base.sm};\n  height: ${({ theme }) => theme.spacing.base.sm};\n  background: ${({ theme, type = 'primary' }) => theme.color[type]};\n  font-family: ${({ theme }) => theme.font.family.default};\n  font-size: ${({ theme }) => theme.font.size.xxs};\n  color: ${({ theme, type = 'primary' }) =>\n    type === 'white' ? theme.color.grey6 : theme.color.white};\n  border-radius: 100%;\n  text-align: center;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n`\n"],"file":"Status.js"}