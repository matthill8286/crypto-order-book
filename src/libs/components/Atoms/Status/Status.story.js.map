{"version":3,"sources":["../../../../src/components/Atoms/Status/Status.story.tsx"],"names":["radios","storiesOf","React","Status","Icon","StyleguideBookmarkOutlined","statusTypes","knobs","type","module","add"],"mappings":";;AAAA,SAASA,MAAT,QAAuB,wBAAvB;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAOC,KAAP,MAAkB,OAAlB;AACA,SAASC,MAAT;AACA,SAASC,IAAT;AACA,SAASC,0BAAT,QAA2C,mCAA3C;AAEA,IAAMC,WAA0C,GAAG;AAAE,KAAG,OAAL;AAAc,KAAG,OAAjB;AAA0B,KAAG;AAA7B,CAAnD;;AAEA,IAAMC,KAAK,GAAG,SAARA,KAAQ,GAAmB;AAC/B,SAAO;AACLC,IAAAA,IAAI,EAAER,MAAM,CAAC,MAAD,EAASM,WAAT,EAAsBA,WAAW,CAAC,CAAD,CAAjC;AADP,GAAP;AAGD,CAJD;;AAMAL,SAAS,CAAC,4BAAD,EAA+BQ,MAA/B,CAAT,CACGC,GADH,CACO,SADP,EACkB,YAAM;AACpB,SAAO,oBAAC,MAAD,EAAYH,KAAK,EAAjB,MAAP;AACD,CAHH,EAIGG,GAJH,CAIO,WAJP,EAIoB,YAAM;AACtB,SACE,oBAAC,IAAD;AAAM,IAAA,KAAK,EAAC,OAAZ;AAAoB,IAAA,MAAM,EAAC;AAA3B,KACE,oBAAC,0BAAD,OADF,EAEE,oBAAC,MAAD,eAAYH,KAAK,EAAjB;AAAqB,IAAA,UAAU;AAA/B,UAFF,CADF;AAQD,CAbH","sourcesContent":["import { radios } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport React from 'react'\nimport { Status, StatusProps, StatusType } from './Status'\nimport { Icon } from '@/components/Atoms/Icon'\nimport { StyleguideBookmarkOutlined } from '@matthill8286/atomic-icon-library'\n\nconst statusTypes: { [key: number]: StatusType } = { 0: 'white', 1: 'black', 2: 'primary' }\n\nconst knobs = (): StatusProps => {\n  return {\n    type: radios('Type', statusTypes, statusTypes[2]),\n  }\n}\n\nstoriesOf('Design System/Atoms/Status', module)\n  .add('Default', () => {\n    return <Status {...knobs()}>1</Status>\n  })\n  .add('With Icon', () => {\n    return (\n      <Icon color=\"grey4\" height=\"md\">\n        <StyleguideBookmarkOutlined />\n        <Status {...knobs()} withinIcon>\n          1\n        </Status>\n      </Icon>\n    )\n  })\n"],"file":"Status.story.js"}