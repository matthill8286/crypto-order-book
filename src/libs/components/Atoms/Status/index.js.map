{"version":3,"sources":["../../../../src/components/Atoms/Status/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './Status'\n"],"file":"index.js"}