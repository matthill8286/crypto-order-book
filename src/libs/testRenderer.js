import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map