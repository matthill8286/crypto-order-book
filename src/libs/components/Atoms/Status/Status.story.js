function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { radios } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { Status } from "./Status";
import { Icon } from "../Icon";
import { StyleguideBookmarkOutlined } from '@matthill8286/atomic-icon-library';
var statusTypes = {
  0: 'white',
  1: 'black',
  2: 'primary'
};

var knobs = function knobs() {
  return {
    type: radios('Type', statusTypes, statusTypes[2])
  };
};

storiesOf('Design System/Atoms/Status', module).add('Default', function () {
  return React.createElement(Status, knobs(), "1");
}).add('With Icon', function () {
  return React.createElement(Icon, {
    color: "grey4",
    height: "md"
  }, React.createElement(StyleguideBookmarkOutlined, null), React.createElement(Status, _extends({}, knobs(), {
    withinIcon: true
  }), "1"));
});
//# sourceMappingURL=Status.story.js.map