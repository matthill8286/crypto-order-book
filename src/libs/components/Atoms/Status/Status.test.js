import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { Status } from "./Status";
describe('Status', function () {
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(Status, {
      type: 'primary'
    }, "1"));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=Status.test.js.map