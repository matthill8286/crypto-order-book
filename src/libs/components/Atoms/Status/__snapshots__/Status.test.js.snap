// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders correctly 1`] = `
<div
  className="Status-y46az8-0 jexwdk"
  type="primary"
>
  1
</div>
`;
